src/common/libflux/stats.h

```diff
@@ -0,0 +1,67 @@
+/************************************************************\
+ * Copyright 2014 Lawrence Livermore National Security, LLC
+ * (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+ *
+ * This file is part of the Flux resource manager framework.
+ * For details, see https://github.com/flux-framework.
+ *
+ * SPDX-License-Identifier: LGPL-3.0
+\************************************************************/
+
+#ifndef _FLUX_CORE_STATS_H
+#define _FLUX_CORE_STATS_H
+
+#include <stdarg.h>
+
+#include "handle.h"
+
+/* Metric types:
+ * Counter - an integer value that will, on the backend (brubeck) send the 
+ *           count and reset to 0 at each flush. It calculates the change 
+ *           from the value sent at the previous flush.
+ * Gauge   - an integer that takes an arbitrary value and maintains its 
+ *           it is set again. Gauges can also take incremental values
+ *           in the form of a + or - in front of the value which increments
+ *           the previously stored value.
+ * Timer   - a double value which represents the time taken for a given
+ *           task in ms.
+ */
+
+/* Format a metric name and increment its value by '1'.
+ */
+int flux_stats_inc (flux_t *h, const char *fmt, ...)
+        __attribute__ ((format (printf, 2, 3)));
+
+/* Format a metric name and decrement its value by '1'.
+ */ 
+int flux_stats_dec (flux_t *h, const char *fmt, ...)
+        __attribute__ ((format (printf, 2, 3)));
+
+/* Update (or create) and store 'count' for 'name' to be sent on the
+ * next flush.
+ */
+int flux_stats_count (flux_t *h, const char *name, ssize_t count);
+
+/* Update (or create) and store 'value' for 'name' to be sent on the
+ * next flush. The'inc' indicates wether or not 'value' is some delta on 
+ * the previous value. If 'inc' is set and 'name' was not previously stored, 
+ * then the value is stored directly.
+ */
+int flux_stats_gauge (flux_t *h, const char *name, ssize_t value, bool inc);
```

Comment:
I think I argued for keeping the `inc` parameter here on the :coffee: call, but I was confused.

I do think there should be a way to increment, decrement, or set a gauge, but I would suggest/prefer 2 separate functions for clarity and ease of use in the code, e.g as a first cut, something like:

```c
int flux_stats_gauge_set (flux_t *h, const char *name, size_t val);
int flux_stats_gauge_inc (flux_t *h, const char *name, ssize_t val);
```

I guess `flux_stats_gauge_inc (h, "mystat", -1)` would then work to decrement a gauge by 1. Optionally, if preferred you could have a separate `flux_status_gauge_dec()`.

As we also talked about it might be nice to provide a "getter", but probably not necessary at this time.

(This means we could probably drop the `flux_stats_inc()` and `flux_stats_dec()` above?)